File: Code/client/src/app/components/card-vehicle/car-park.ts
import { Vehicle } from './vehicle';

export class CarPark {

  public vehicle: Vehicle;
  public entryDate: string;
  public slotNumber: number;
  public notes: string;

  constructor() {
    this.vehicle = new Vehicle();
  }

  public get available(): boolean {
    const plate = this.vehicle.plate || '';
    return plate.length < 6;
  }


  public copy(carPark: CarPark) {
    this.vehicle.copy(carPark.vehicle);
    this.entryDate = carPark.entryDate;
    this.slotNumber = carPark.slotNumber;
    this.notes = carPark.notes;
  }

  public restore(): void {
    this.vehicle.plate = null;
    this.vehicle.cylinder = null;
    this.entryDate = null;
    this.notes = null;
  }

}

File: Code/api/src/main/java/co/com/ceiba/parking/domain/CarPark.java
/**
 * 
 */
package co.com.ceiba.parking.domain;

import java.time.LocalDateTime;

import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonIgnore;

import co.com.ceiba.parking.entities.DbCarPark;

/**
 * Domain bean of the entity {@link DbCarPark}.
 * 
 * @author fabian.diaz
 *
 */
public final class CarPark {

  private long id;

  @NotNull(message = "El vehiculo no puede ser nulo")
  private Vehicle vehicle;
  private LocalDateTime entryDate;
  private LocalDateTime exitDate;
  private short slotNumber;
  private String notes;

  public long getId() {
    return id;
  }

  public void setId(long id) {
    this.id = id;
  }

  public Vehicle getVehicle() {
    return vehicle;
  }

  public void setVehicle(Vehicle vehicle) {
    this.vehicle = vehicle;
  }

  public LocalDateTime getEntryDate() {
    return entryDate;
  }

  public void setEntryDate(LocalDateTime entryDate) {
    this.entryDate = entryDate;
  }

  public LocalDateTime getExitDate() {
    return exitDate;
  }

  public void setExitDate(LocalDateTime exitDate) {
    this.exitDate = exitDate;
  }

  public short getSlotNumber() {
    return slotNumber;
  }

  public void setSlotNumber(short slotNumber) {
    this.slotNumber = slotNumber;
  }

  public String getNotes() {
    return notes;
  }

  public void setNotes(String notes) {
    this.notes = notes;
  }

  /**
   * Convert domain bean to entity.
   * 
   * @return Entity bean.
   */
  @JsonIgnore
  public DbCarPark toEntity() {
    DbCarPark dbCarPark = new DbCarPark();
    dbCarPark.setId(getId());
    dbCarPark.setVehicle(getVehicle() == null ? null : getVehicle().toEntity());
    dbCarPark.setEntryDate(getEntryDate());
    dbCarPark.setExitDate(getExitDate());
    dbCarPark.setSlotNumber(getSlotNumber());
    dbCarPark.setNotes(getNotes());
    return dbCarPark;
  }

}
